snorkel/classification/training/trainer.py

```diff
@@ -509,3 +510,66 @@ def _reset_losses(self) -> None:
         """Reset the loss counters."""
         self.running_losses = defaultdict(float)
         self.running_counts = defaultdict(int)
+
+    def save(self, trainer_path: str) -> None:
+        """Save the trainer config to the specified file path in json format.
+
+        Parameters
+        ----------
+        trainer_path :
+            The path where trainer config and optimizer state should be saved.
+        """
+
+        head, tail = os.path.split(trainer_path)
+
+        if not os.path.exists(head):
+            os.makedirs(os.path.dirname(head))
+        try:
+            torch.save(
+                {
+                    "trainer_config": self.config._asdict(),
+                    "optimizer_state_dict": self.optimizer.state_dict(),
+                },
+                trainer_path,
+            )
+        except BaseException:  # pragma: no cover
+            logging.warning("Saving failed... continuing anyway.")
+
+        logging.info(f"[{self.name}] Trainer config saved in {trainer_path}")
+
+    def load(
+        self, trainer_path: str, model: Optional[MultitaskClassifier] = None
+    ) -> None:
+        """Load trainer config and optimizer state from the specified json file path to the trainer object. The optimizer state is stored, too. However, it only makes sense if loaded with the correct model again.
```

Comment:
can we add an example for expected usage, loading with with the corresponding model + resuming training? (see https://github.com/snorkel-team/snorkel/blob/master/snorkel/labeling/model/label_model.py#L366-L372 for an example, of examples (😄) in docstrings)